src/irmin-pack/pack.ml

```diff
@@ -191,6 +191,17 @@ struct
         true )
       else false
 
+    let flush t =
+      Dict.flush t.pack.dict;
+      IO.flush t.pack.block;
+      Index.flush t.pack.index;
+      Tbl.clear t.staging
+
+    let set_hook_index index t =
+      Index.set_merge_hook index (fun () ->
+          Dict.flush t.pack.dict;
+          IO.flush t.pack.block)
```

Comment:
Fsync needed here?

Reply:
I'm thinking this is maybe not necessary. We are not using `fsync` for any pack flush, but instead we use the `integrity_check` to repair the pack/index. I think `fsync` is costly, and as we have the `integrity_check` already, we don't need it. What do you think?

Reply:
As discussed offline, catching it with `integrity_check` is probably the way to go to avoid having costly flushes. There's now a comment here to explain this intent :+1: